lib/syskit/network_generation/engine.rb

```diff
@@ -537,7 +537,8 @@ def reconfigure_tasks_on_static_port_modification(deployed_tasks)
                     work_plan
                     .find_tasks(Syskit::TaskContext).not_finished.not_finishing
                     .find_all do |t|
-                        deployed_tasks.include?(t) && (t.setting_up? || t.setup?)
+                        deployed_tasks.include?(t) &&
+                        (t.setting_up? || (t.setup? && !t.read_only?))
```

Comment:
@doudou I'm not sure why this was needed, but it solved an error that was occuring. It stated that in line 545, there was no `transaction.modifies_static_ports?` for `nil` class. Any idea if this is the correct solve?